components/feature/p2p/src/main/java/mozilla/components/feature/p2p/view/P2PBar.kt

```diff
@@ -0,0 +1,180 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.p2p.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.Button
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.constraintlayout.widget.ConstraintLayout
+import kotlinx.android.synthetic.main.mozac_feature_p2p_view.view.*
+import mozilla.components.feature.p2p.R
+
+private const val DEFAULT_VALUE = 0
+
+/**
+ * A toolbar for peer-to-peer communication between browsers. Setting [listener] causes the
+ * buttons to become active.
+ */
+@Suppress("TooManyFunctions")
+class P2PBar @JvmOverloads constructor(
```

Comment:
A lot of this logic should be removed from the View and put into a presenter (See other examples of interactor/presenter patterns in the codebase). We want our Android Views to do just the one thing: displaying the UI. If there is logic how to decide what to show we should put that in a presenter. TabsTrayPresenter and ToolbarPresenter are other examples.

Reply:
@jonalmeida I'm looking again at this code while my teammates are away. I know you are too, so I won't expect a quick reply.

I want to make sure I understand how Presenter works with MVC. Would the presenter mediate all of the interactions between the view and the controller? `V <--> P <--> C`